source/features/unfinished-comments.tsx

```diff
@@ -1,26 +1,34 @@
 import select from 'select-dom';
+import delegate from 'delegate-it';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
 let documentTitle: string | undefined;
 
 function hasDraftComments(): boolean {
-	return select.all('textarea').some(textarea => textarea.value.length > 0 && textarea.offsetWidth > 0);
+	return select.all('textarea').some(textarea => textarea.value.length > 0 && textarea.offsetWidth > 0 && !textarea.closest('.rgh-is-sending-comment'));
 }
 
-async function updateDocumentTitle(): Promise<void> {
-	if (documentTitle) {
-		document.title = documentTitle;
-		documentTitle = undefined;
-	} else if (document.visibilityState === 'hidden' && hasDraftComments()) {
+function updateDocumentTitle(): void {
+	if (document.visibilityState === 'hidden' && hasDraftComments()) {
 		documentTitle = document.title;
 		document.title = '(Draft comment) ' + document.title;
+	} else if (documentTitle) {
+		document.title = documentTitle;
+		documentTitle = undefined;
 	}
+
+	select('.rgh-is-sending-comment')?.classList.remove('rgh-is-sending-comment');
+}
+
+function tagSendingForm({delegateTarget}: delegate.Event<Event, HTMLElement>): void {
+	delegateTarget.classList.add('rgh-is-sending-comment');
```

Comment:
This is the right time to remove the class:

```suggestion
function tagSendingForm({delegateTarget: form}: delegate.Event<Event, HTMLFormElement>): void {
	form.classList.add('rgh-is-sending-comment');
	form.addEventListener('reset', () => {
			form.classList.remove('rgh-is-sending-comment');
	}, {once: true})
```
Untested, likely not supported:

- review comments
- editing comments
- PR submissions